### ISSUE TITLE
Fix `got.mergeOptions` docs

### ISSUE THREAD

**Issue by Gamaranto** [sindresorhus/got#1063]
#### Describe the bug

 - Node.js version: 
 - OS & version: 

#### Actual behavior
The objects get merged like with a normal spread operation `{...defaultOptions, ...newOptions}`
...

#### Expected behavior
The result of `got.mergeOptions` should be `{foo: 'x', bar: 'y'}`
...

#### Code to reproduce

```js
import got, { Options } from 'got';

const defaultOptions: Options = {
  responseType: 'json',
  json: {
    foo: 'x',
  },
};

const newOptions: Options = {
  json: {
    bar: 'y',
  },
};

console.log(got.mergeOptions(defaultOptions, newOptions).json);
// {bar: 'y'}
console.log(got.mergeOptions(newOptions, defaultOptions).json);
// {foo: 'x'}
```

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment]
This is actually the correct behavior. It's like you would want to merge `body: 'a'` and `body: 'b'` and expect `body: 'ab'` - these are two different payloads.

**Comment by szmarczak** [comment]
The documentation needs to be fixed

**Comment by goto1** [comment]
It sounds like this statement in the documentation is not quite correct, meaning it has some caveats:

> - If the new property is a plain `object`:
>   - **If the parent property is a plain object too, both values are merged recursively into a new object.**

So for example, `form`, `json`, and `body` wouldn't get "merged" but instead it would use values from properties coming from the `newOptions` object, is that correct?

**Comment by szmarczak** [comment]
yup

**Comment by goto1** [comment]
Got it. I could take a look and have a crack at it.

**Comment by szmarczak** [comment]
Fixed in 157e02bcb613ab506829326e7af96e8155b4e186
